cron/test_projects.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-console */
+/**
+ * Test expiration date of user, if lower than 2 month, send an email to user
+ */
+
+const { promisify } = require('util');
+const sleep = promisify(setTimeout);
+
+const cfgsrv = require('../core/config.service.js');
+let my_conf = cfgsrv.get_conf();
+const CONFIG = my_conf;
+
+const dbsrv = require('../core/db.service.js');
+const plgsrv = require('../core/plugin.service.js');
+const maisrv = require('../core/mail.service.js');
+
+const MAILER = CONFIG.general.mailer;
+
+const notif = require('../core/notif_'+MAILER+'.js');
+
+function timeConverter(tsp){
+    let a = new Date(tsp);
+    let months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'];
+    let year = a.getFullYear();
+    let month = months[a.getMonth()];
+    let date = a.getDate();
+    let hour = a.getHours();
+    let min = a.getMinutes();
+    let sec = a.getSeconds();
+    return date + ',' + month + ' ' + year + ' ' + hour + ':' + min + ':' + sec ;
+}
+
+dbsrv.init_db().then(async ()=>{
+    plgsrv.load_plugins();
+    let projects = await dbsrv.mongo_projects().find({expire: {$lt: (new Date().getTime() + 1000*3600*24*60)}}).toArray();
+
+    let mail_error = 0;
+    if (! notif.mailSet()){
+        console.log('Error: mail is not set');
+        process.exit(1);
+    }
+
+    let in60Days = new Date().getTime() + 1000*3600*24*60;
+    let in15Days = new Date().getTime() + 1000*3600*24*15;
+    let in3Days = new Date().getTime() + 1000*3600*24*3;
+
+    for(let i=0;i<projects.length;i++){
+        let project = projects[i];
+        if (project.expiration_notif === undefined) {
+            project.expiration_notif = 0;
+        }
+        let do_notify = false;
+        if (project.expiration_notif == 0 && (project.expire <= in60Days)) {
+            project.expiration_notif = 1;
+            do_notify = true;
+        } else if (project.expiration_notif == 1 && (project.expire <= in15Days)) {
+            project.expiration_notif = 2;
+            do_notify = true;
+        } else if (project.expiration_notif == 2 && (project.expire <= in3Days)) {
+            project.expiration_notif = 3;
+            do_notify = true;
+        }
+        if (!do_notify) {
+            continue;
+        }
+        console.log(`Project will expire, send notication number ${project.expiration_notif} to ${project.id}`);
+        let user = await dbsrv.mongo_users().findOne({uid: project.owner});
```

Comment:
should manage case where user is not found

Reply:
701f4411fa1e46e0b91a3983e1a3048bda268613 Done, and I also add a check on project owner